### PR TITLE
Show a preview of the control in the SUI

### LINKED ISSUE #9122: Add a preview of your terminal to the Settings UI
# Description of the new feature/enhancement

[Design doc link](https://github.com/microsoft/terminal/blob/main/doc/specs/%231564%20-%20Settings%20UI/design.md#profile-appearance-page)

It would be really cool if we had a preview of the terminal in the Settings UI > Profile > Appearance page and the Settings UI > Color Schemes page.

Implementation plan:
- Move `TerminalSettings` to `TerminalSettingsModel` project
- Introduce 
- In Settings UI...
  - embed `TermControl` where you want it
  - generate a `TerminalSettings` object for `TermControl` initialization (see #8602 )
  - (Profiles) introduce a `TerminalConnection` to use here
  - (Color Scheme) introduce a `TerminalConnection` with the color tool color table

### PR DESCRIPTION
**Pull request by PankajBhojwani**
In the 'Appearance' tab of a profile, show a preview of what the control looks like

## PR Checklist
* [x] Closes #9122 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [x] I work here

### CODE REVIEW

**Comment by zadjii-msft** [comment]
> * [ ]  We probably want more things in `PreviewConnection`
>

Thoughts:
* We could just go with the text that was in the old propsheet preview:
  image
  but we could probably do better than that
* We could do the whole color table, 
  image
  but that requires 80 columns to display right, which is a lot. Plus that's maybe better suited for the color scheme page
* What if we had something simple with default, red, and green text in it, like some git output? 
  image

**Comment by zadjii-msft** [comment]
Other random thought - should the preview be _above_ the scroll viewer, so the preview is always visible as you scroll through the list of settings 🤔? @cinnamon-msft for design help here

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Great work! Concerned about a few scenarios that I want to make sure we handle:
- **Resizing**
  - Please resize the window and make sure it feels natural. We'll want some padding on the left and right side so that the TermControl is always visible.
  - If possible, reflow would be awesome. But I bet that's hard. I'm ok with that being a follow-up, if we even want to pursue it.
  - Make sure to test the smallest possible size of the window.
- **Interactivity**
  - Can you select text in the TermControl? Originally, I thought we should. But we wouldn't be able to copy/paste, right? So maybe we should disable selection, and if people are interested, throw in selection/copy/paste via the mouse later?
- **Misc settings**
  - Do the following profile settings do anything?
    - Retro terminal effects
    - Cursor shape & height
    - Acrylic and acrylic opacity
    - scrollbar visibility
- **Accessibility**
  - Can you tab in/out of the TermControl?
  - What does the screen reader say when you are focused on the TermControl? (I can probably check this one after the PR is out of draft)
- **Preview Connection**
  - Probably better for a future consideration, but it would be cool if we could have the header for the shell pop up. Like, CMD would do the "Microsoft Windows [Version..." and Pwsh would do "PowerShell 7.1.3 Copyright". But we'd need a way to figure out what the output is and that just sounds messy.
  - I'd also like it if we could get powerline in there, since that's really colorful. But again, that doesn't sound possible tbh.
  - Honestly, I'm tempted to say that we should just open a real terminal connection in there? It might be overkill, but it would allow users to display whatever they want. That said, I think I'm ok with some simple output like this for now, then doing a more complex one in a follow-up task.

**Comment by DHowett** [comment]
> real connection

> snippet from the shell

**ABSOLUTELY NOT**.

Imagine I make a profile that runs...

`wsl -d Ubuntu rm -rf /home/dustin/profile_launched` or whatever?
I'm typing it in, and I get to `wsl -d Ubuntu rm -rf /home` and well, there you go. Terminal decided to RUN THE COMMAND just to give me, what, a preview?

We don't want to spawn TWO NEW PROCESSES for every preview update, either.

The connection needs to be _representative_ of what people use Terminal for. Powerline is not guaranteed (in the font); a shell is, emoji may be, etc.

I'm imagining that we have emoji in there, and some characters that _might_ ligature together, and some colors. Not a lot of them (check the color scheme page for that). The emoji will change when you toggle colored emoji, etc.

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/Profiles.xaml:218]: If only we had a normal contructor for the control, we could use `<tc:TermControl Height="200"...`

File: src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp:42]: Reflow will be impossible. On resize, the connection should simply clear the terminal and _re-print the entire preview text_.

File: src/cascadia/TerminalSettingsEditor/PreviewConnection.h
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/PreviewConnection.h:11]: Here is the yardstick for whether something should be in the TC project:

1. Is it something _every consumer of TC should have access to?_
2. Is it worth putting on the public API of TC?

I suspect the answers are no and no.

File: src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp:42]: The reason for this is that resize is a destructive operation when content is pushed off the top of the screen.

**Comment by cinnamon-msft** [comment]
> Other random thought - should the preview be _above_ the scroll viewer, so the preview is always visible as you scroll through the list of settings 🤔? @cinnamon-msft for design help here

Yeah I think this makes sense. I think it'll look okay once we change the Pivot to the horizontal NavView.

**Comment by mdtauk** [comment]
> > Other random thought - should the preview be _above_ the scroll viewer, so the preview is always visible as you scroll through the list of settings 🤔? @cinnamon-msft for design help here
> 
> Yeah I think this makes sense. I think it'll look okay once we change the Pivot to the horizontal NavView.

You may experience weirdness nesting a NavigationView inside another NavigationView

**Comment by PankajBhojwani** [comment]
> Misc settings
> Do the following profile settings do anything?
> 
> - Retro terminal effects
> - Cursor shape & height
> - Acrylic and acrylic opacity
> - scrollbar visibility

They all work except acrylic!

**Comment by carlos-zamora** [comment]
> They all work except acrylic!

I'm a bit conflicted on this. Definitely not blocking the PR for this, but we should probably file a follow-up to be able to provide a preview of the acrylic (if that's even possible).

But if it's not too much work for you to do it now, that'd be cool too ;) haha.

**Comment by PankajBhojwani** [comment]
A few updates:

- The preview connection now automatically sends the default text, as well as red/green/blue text
- The preview connection no longer allows _any_ input from the user
- The control preview is now above the General/Appearance/Advanced tabs (so this automatically gives us the ability to scroll without losing sight of the control)
- When the window width is <640 px, the control preview will resize itself to fit the window width (with a small margin on both sides), otherwise the control preview width is set at 640

**Comment by PankajBhojwani** [comment]
> Can you tab in/out of the TermControl?
> What does the screen reader say when you are focused on the TermControl? (I can probably check this one after the PR is out of draft)

The preview term control is not focusable and cannot be tabbed in/out of! Also the height of the preview has been reduced by half so that the scroller below the pivot has more space. (can't post a screenshot because of secrets)

**Comment by carlos-zamora** [comment]
## Notes from Demo w/ @cinnamon-msft
- Add a border to the preview (probably with the accent color?)
  - This'll help distinguish the preview background from the page background.
- Cursor should just stay on (no blinking)
  - This'll make it look like it's not interactive.
- Move terminal preview to be under the pivot control/selector
  - The space between the page header and the preview looks a little weird. This might fix that problem and make it look more natural.
- Move the preview to be inside the "Appearance" pivot item
  - There's no need for us to be able to see the preview when in "General" or "Advanced", really. The entire purpose of this is to be a preview of the appearance, so it'll look nice if it's _only_ in the "Appearance" item.
- Align the preview to the left
  - I disagree with Kayla on this one, but who knows, maybe it'll look good. She has a better eye for this stuff than I do, generally.



### Content inside the preview
- People probably want to see something more like a prompt. Kayla and I came up with this...
```
Windows Terminal
Copyright (c) Microsoft Corporation.

C:\Windows\Terminal>  <CURSOR>
```
I think it'll look nice and kinda match the initial output of PowerShell/CMD.

- [**punt**] figure out a way to show selected text. Some ideas:
  - let you select text (I feel like this is the least appealing option here)
  - have some pre-selected text on the preview?
- [**punt**] show colored text:
  - This one is weird. The preview in the ColorSchemes page is going to show this off a lot better with a full color table. So I think we should just revisit this topic once we get a preview in the ColorSchemes page
  - I'm not really too sure what the best approach here is, so that's why I think it's better to punt on it and get this in.
- [**punt + feature idea?**] Allow multiple customizable previews:
  - Realistically, we'll never be able to make everybody happy, especially when you start taking into account that some people have a powerline theme (and it might be a custom one too). Maybe we should eventually allow users to import their own preview output? Like, they can import some kind of text file that is just gets output via the preview connection, and any color changing VT sequences get handled too?
  - This would be a _huge_ undertaking and is definitely off topic, but I think this is a good example of future work that we can do after this feature goes in and we get some user feedback. 😊


### Miscellaneous notes
- "scrollbar visibility" doesn't do anything
  - It makes sense that it doesn't. We only show it if we're scrolled down.
- good job on the scrolling behavior
  - the preview should always be visible, so it's nice that you can just make changes and _immediately_ see them without having to scroll up. _LOVE IT!_
- Acrylic works!!!!

**Comment by DHowett** [comment]
> ## Notes from Demo w/ @cinnamon-msft Kayla Cinnamon FTE
> * Add a border to the preview (probably with the accent color?)
>   
>   * This'll help distinguish the preview background from the page background.
> * Cursor should just stay on (no blinking)
>   
>   * This'll make it look like it's not interactive.
> * Move terminal preview to be under the pivot control/selector
>
>   * The space between the page header and the preview looks a little weird. This might fix that problem and make it look more natural.

This is incompatible with keeping it visible while you scroll, which you loved.

> * Move the preview to be inside the "Appearance" pivot item
>   
>   * There's no need for us to be able to see the preview when in "General" or "Advanced", really. The entire purpose of this is to be a preview of the appearance, so it'll look nice if it's _only_ in the "Appearance" item.

What about Advanced > Text Antialiasing, Advanced > Bell Notification Style, General > Tab Title, General > Tab Color (when we have tab preview), General > Icon?

> * Align the preview to the left
>   
>   * I disagree with Kayla on this one, but who knows, maybe it'll look good. She has a better eye for this stuff than I do, generally.

I think this will look bad.

> 
> ### Content inside the preview
> * People probably want to see something more like a prompt. Kayla and I came up with this...
> 
> ```
> Windows Terminal
> Copyright (c) Microsoft Corporation.
> 
> C:\Windows\Terminal>  <CURSOR>
> ```
> 
> I think it'll look nice and kinda match the initial output of PowerShell/CMD.

The initial output of PowerShell/CMD is atrocious, and I see no reason for us to make that mistake again. We've been showing people how to customize
their shells for two years now, and it seems shortsighted for us to display a preview of what it used to look like in 1985.

We need to display, at the minimum: Colors. Emoji (because we will have a [disable emoji] [(x) use color emoji] switch, per user request). Italic text. Bold text (when we have bold.)
I don't care if Color is covered by the Color Schemes preview: it doesn't show up oiver a background image or over acrylic, because those _aren't part of a color scheme_.

All of those are not represented in the shell rpeamble.

> 
> ...
>   
> * [**punt + feature idea?**] Allow multiple customizable previews:
>   
>   * Realistically, we'll never be able to make everybody happy, especially when you start taking into account that some people have a powerline theme (and it might be a custom one too). Maybe we should eventually allow users to import their own preview output? Like, they can import some kind of text file that is just gets output via the preview connection, and any color changing VT sequences get handled too?
>   * This would be a _huge_ undertaking and is definitely off topic, but I think this is a good example of future work that we can do after this feature goes in and we get some user feedback. 😊

Hard veto. Absolutely not. There's no purpose apart from making ourselves feel accomplished.
We should make the preview as inclusive of normal person scenarios as we possibly can.

> 
> ...
> 
> * good job on the scrolling behavior
>   
>   * the preview should always be visible, so it's nice that you can just make changes and _immediately_ see them without having to scroll up. _LOVE IT!_

This is incompatible with moving the preview under the pivot, as you suggested.

_/cc @cinnamon, since some of the refuted opinions might have been hers_

**Comment by carlos-zamora** [comment]
### Move terminal preview to be under the pivot control/selector

This should be possible by doing the following:
```xml
<PivotItem x:Uid="Profile_Appearance">
	<Grid>
    	<Grid.RowDefinitions>
			<RowDefinition Height="Auto"/>
			<RowDefinition Height="*"/>
		</Grid.RowDefinitions>

		<Border x:Name="ControlPreview"
				Height="320"
            	Margin="8,0,8,0"
            	HorizontalAlignment="Stretch"
            	Grid.Row="0"/>

		<ScrollViewer Grid.Row="1">
			<StackPanel>
				<!-- All of the settings controls -->
		</ScrollViewer>
	</Grid>
</PivotItem
```
Tested it out myself, and that doesn't break the scrolling.

Now, the only question left is...
> What about Advanced > Text Antialiasing, Advanced > Bell Notification Style, General > Tab Title, General > Tab Color (when we have tab preview), General > Icon?

This'll require a broader discussion on the design. Let's talk about this at our next sync meeting.

### Align the preview to the left

I agree that this'll look bad too (sorry Kayla haha). But it should be a quick one-line change.

@PankajBhojwani mind making that change and sending it over to @cinnamon-msft (or even the team over email if you want). I think it's just a matter of visualizing it tbh.

### Content inside the preview

This'll require a broader discussion on the design. Let's talk about this at our next sync meeting.

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Honestly, it seems like most of the issues left are design-related.

Blocking for two reasons...
1. we need consensus on the design (we'll talk about that in the upcoming sync meeting)
2. resolution on using `TerminalSettings::ApplyProfileSettings` instead of `TermControl::Settings` setter

Also, don't forget to update the PR description.

File: src/cascadia/TerminalControl/TermControl.h
Comment (DHowett) [reply src/cascadia/TerminalControl/TermControl.h:117]: I don't necessarily think this should be part of TS's public API.

Comment (carlos-zamora) [reply src/cascadia/TerminalControl/TermControl.h:117]: Eh, fine. But I'm curious, why not?

File: src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw:438]: These might need explanatory comments

File: src/cascadia/TerminalControl/TermControl.h
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/TermControl.h:117]: I'm firmly opposed to the idea that TermControl has any concept of a Profile. It should be given a fully-constructed `IControlSettings`, and it's up to the application hosting the control to decide how to populate that object.

I agree that exposing the Settings setter is definitely a footgun. I don't have a better idea right now though...

**Review by zadjii-msft — COMMENTED** [review]
TBH, I don't care so much to block over the center -> left align change. Making the cursor visible might be important though - not sure how to force that easily. IMO, just leave the text for now, and fix it in post. That's _easy_ to change. 

Also, go back and bump the OP - it's unclear what's still a todo and what's a to-done

**Comment by DHowett** [comment]
I seriously think the text should not be localizable. I don't want to ship with it localizable and then have to back that out. :smile:

File: src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp:31]: > This method still needs to be implemented for that right?

Ah it actually doesn't! The connection doesn't need to do anything

File: src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw:438]: Removed it since we no longer want localizable text for this

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Profiles.xaml:801]: Omg you are completely right, we don't need this visual state manager at all! Thanks :)

File: src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp:22]: very clever. we do also remove the cursor when it loses focus, though, too.

Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp:22]: Ah, the preview control is no longer enabled nor focusable (see Profiles.cpp lines 484-485) - so the cursor just stays there and with this sequence it doesn't blink

**Comment by PankajBhojwani** [comment]
Updates:

- There is now a little tab above the control preview, where the icon and title will update accordingly (would like feedback on this if someone wants to switch to this branch and play with it)
- The preview is now left aligned (it will be extremely annoying to align the fake tab with the preview without doing this)
- The cursor is now always on (doesn't blink)

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Profiles.xaml]: Had to add this back in because we now change the horizontal alignment with this, and for some reason that causes the preview to disappear into oblivion unless we also set the width

File: src/cascadia/TerminalSettingsEditor/Profiles.cpp
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Profiles.cpp:428]: when the Name or TabTitle fields change (make sure you listen to PropertyChanged notifications), you need to tell xaml that TabDisplayName changed

Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Profiles.cpp:432]: Ah of course, thanks for catching that!

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Profiles.xaml:71]: Feels weird to have a 1-column wide grid. Could you remove the `Grid` and add `Grid.Row=0` and `Margin="8,0,8,0` to the `Border` below?

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Profiles.xaml:106]: If you're adding `Margin="8,0,8,0"` to all of the children of `Grid`, move `Margin` to be on the `Grid` instead.

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Profiles.xaml:72]:
```suggestion
                <!-- Tab Preview -->
                <Border MinWidth="128"
```
or some kind of comment. Took me a sec to understand that this whole thing is creating a fake tab.

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Profiles.xaml:107]: dumb question: what happens if we just _always_ have this set as `HorizontalAlignment="Left"`?

File: src/cascadia/TerminalSettingsEditor/Profiles.cpp
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Profiles.cpp:430]:
```suggestion
    hstring ProfileViewModel::TabDisplayName() const
```
and in the .h too

File: src/cascadia/TerminalSettingsEditor/Profiles.cpp
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Profiles.cpp:448]:
```suggestion
        _previewControl{ Control::TermControl(Model::TerminalSettings{}, make<PreviewConnection>()) }
```
nit: `()` makes it ambiguous if you're calling a function or a ctor sometimes. Dustin scared me into this one ages ago hahaha

**Review by carlos-zamora — APPROVED** [review]
This is great! Approving with some small fixes.

Please file a follow-up task to fix the following:
- The layout still feels a bit tight at times. I'd like to see...
   1. list of settings reflows into 2 columns when we have enough space
      - I don't really like that you can only see like 1 or 2 settings at once if the window is too small. The preview taking up some space makes this more of a problem.
   2. Make the preview hide-able
      - I think we could play around with this a bit and make it so that if the preview takes up half of the window (or some other heuristic), the preview is hidden behind a "Show/Hide Preview" toggle button (or even just hide it and don't let the user see it unless there's enough space on the page).
      - @cinnamon-msft suggests that we _always_ make that button visible so users can disable the preview if they know what they're doing.

I would still like more details from you/others on why we're going with a setter on `TermControl::Settings` instead of `TerminalSettings::ApplyProfileSettings` being exposed and used. That still seems like the way to go for me, but I don't want to be a pain and block a good PR because of that.

**Comment by cinnamon-msft** [comment]
Here's a screenshot as of 4/19:
image

**Review by cinnamon-msft — CHANGES_REQUESTED** [review]
After playing with this for a bit, I have some concerns with usability and am recommending some changes.

- I strongly believe this should move underneath the Appearance pivot. I know there are appearance settings on the General and Advanced tabs, but I don't think they necessarily need to be shown in the preview window.
- With the above point, I'm not totally convinced we need a tab above the preview. It would look more streamlined without it and I think people can understand what their tab color/tab title would look like without it.
- The whole page should be scrollable so people can look at > 2 settings at once.*
- The width should be the same width as the pivot menu or the width of the text boxes + browse buttons.

I would be nervous shipping this in the state it's currently in and would prefer we fix it in this PR rather than creating follow-up tasks.

* I took a look at Windows settings and there's a preview window in there as well and the whole page is scrollable.
image

**Comment by PankajBhojwani** [comment]
- The preview has moved to under the appearance pivot
- The preview is now inside the Scroll Viewer (and is scrollable)
- The fake tab has been removed
- The width has been made smaller to align with the pivot menu

I think if we want to have the fake tab/settings from the other pivots in the preview, we can figure out how to do it in a clean manner later on. For now though, I've removed all those extra things to try and get this in before the bug bash

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Profiles.xaml]: resolving this and all the other grid-related comments since the grid has been removed!

**Comment by cinnamon-msft** [comment]
This looks great. Probably just want to fix the width, since it's the size of the window. :)
image

**Comment by PankajBhojwani** [comment]
> This looks great. Probably just want to fix the width, since it's the size of the window. :)
> image

Thats _very_ strange, on my machine the width is the size of the pivot headers...

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsEditor/Profiles.xaml:36]: looks unused? leftover from a previous commit maybe?

File: src/cascadia/TerminalSettingsEditor/Profiles.h
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsEditor/Profiles.h:54]: IMO, this comment isn't adding anything lol.
```suggestion
```

**Review by zadjii-msft — APPROVED** [review]
Yea I dig it. IMO, we can fix the polish in post

**Comment by cinnamon-msft** [comment]
I just rebuilt it and it's still expanding the width to the whole window, but I won't block on it and we can fix the bug in a follow-up.

**Review by cinnamon-msft — APPROVED** [review]

**Review by DHowett — CHANGES_REQUESTED** [review]
I'm seeing a crash -- blocking for bug bash

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Profiles.xaml:36]: Ah yes, thanks for the catch!

File: src/cascadia/TerminalSettingsEditor/Profiles.h
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Profiles.h:54]: Ah this reminded me that the entire tab display name stuff needs to be removed, thanks!

**Comment by PankajBhojwani** [comment]
> I am a bit concerned about using the TSF control's settings size here? We're a control -- we have a size of our own.

We use the control's width and height now!

> What happens when the window is resized?

Nothing, the control's width and height are fixed

> Please avoid hardcoding colors. 

We use a system theme resource colour now :)

File: src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp:33]: Wait, no.. on resize, the connection must re-print the entire preview text. Resize with reflow *destroys* the content that goes off the top of the screen, and the connection needs to clear the screen and put it back. Now, since we never resize the control this isn't going to be a problem... but do keep it in mind.

File: src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp:11]:
```suggestion
static constexpr std::wstring_view PreviewText{ L"Windows Terminal\r\nCopyright (c) Microsoft Corporation\r\n\nC:\\Windows\\Terminal> " };
```

**Review by DHowett — APPROVED** [review]
Future design notes:

Having the TermSettings come out of the viewmodel is a good idea. If you embrace this, you can make the viewmodel _send a settings change notification_ for `TermSettings`, and you can delete all of the `UpdateSettings` calls from Profiles itself. This would be letting the VM do the work for you.

For now, though: I'm going to apply my comments and merge this. Excellent work, and thank you for doing it!

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Profiles.xaml]: As long as this.. uh, looks fine. I guess. It's a weird one to use-- we're not a toggle button :smile:

Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Profiles.xaml]: This is an older commit! It uses `SystemControlForegroundBaseMediumLowBrush`

**Pull request merged**
